.github/workflows/common.yml

```diff
@@ -70,26 +75,62 @@ jobs:
       - name: Check build
         run: make release
 
-      - name: Run required checks/linters
+      - name: Run check-license
+        run: |
+          # go run ../.github/check-license.go # TODO: This repo has multiple licenses, fix checker. https://jira.percona.com/browse/PMM-10118
+
+      - name: Run go-sumtype
+        run: bin/go-sumtype ./...
+
+      - name: Run golangci-lint
         env:
-          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COMMAND: "bin/golangci-lint run -c=.golangci-required.yml --out-format=line-number"
+          REDIRECT: "| bin/reviewdog -f=golangci-lint -reporter=github-pr-review -fail-on-error=true"
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.ROBOT_TOKEN || secrets.GITHUB_TOKEN }}
         run: |
-          # go run ../.github/check-license.go # TODO: This repo has multiple licenses, fix checker
-          bin/go-sumtype ./...
-          # use GITHUB_TOKEN because only it has access to GitHub Checks API
-          bin/golangci-lint run -c=.golangci-required.yml --out-format=line-number | bin/reviewdog -f=golangci-lint -reporter=github-pr-review -fail-on-error=true
+          if out=$( ${{ env.COMMAND }} ); exit_code=$?; [ $exit_code -ne 0 ]; then
+            if [ $exit_code -gt 1 ] || ${{ github.event.pull_request == null }}; then
+              echo "$out"
+              exit $exit_code
+            else
+              echo "$out" ${{ env.REDIRECT }}
+            fi
+          else
+            echo "$out"
+          fi
```

Comment:
Later let's replace it with https://github.com/reviewdog/action-golangci-lint

Reply:
In this case we'd need to take care of versions matching in makefile and ci. Will not that be a problem?

Reply:
Can you please create ticket, let's think about that later. Docs saying that in this action they always using latest release, maybe it will work ok.

Reply:
https://jira.percona.com/browse/PMM-10445